### PR TITLE
vtbench: add initial version of a vitess benchmarking tool

### PR DESCRIPTION
**Pull request by demmer**
## Description
Implement a simple benchmark tool that is capable of running queries using either mysql or grpc protocol to a vtgate, or by grpc straight to a vttablet.

## Motivation
Although plenty of off the shelf benchmarking tools existed, I wanted one where I was able to control the parallelism and make apples to apples comparisons between mysql / grpc vtgate / grpc vttablet.

### CODE REVIEW

File: go/cmd/vtbench/vtbench.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	flag.Parse()
+
+	clientProto := vtbench.MySQL
+	switch *protocol {
+	case "", "mysql":
+		clientProto = vtbench.MySQL
+	case "grpc-vtgate":
+		clientProto = vtbench.GRPCVtgate
+	case "grpc-vttablet":
+		clientProto = vtbench.GRPCVttablet
```
Comment (rafael) [inline go/cmd/vtbench/vtbench.go]: Is this to run the test directly against the tablet? We should include it in the protocol list doc string.

File: go/vtbench/vtbench.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+		} else {
+			b.Rows.Add(int64(result.RowsAffected))
+		}
+
+	}
+
+	b.wg.Done()
+}
+
+/*
```
Comment (rafael) [inline go/vtbench/vtbench.go]: Should we remove this commented code?

File: go/cmd/vtbench/vtbench.go (hunk 1 above)
Comment (demmer) [reply go/cmd/vtbench/vtbench.go]: Good idea. Done

File: go/vtbench/vtbench.go (hunk 2 above)
Comment (demmer) [reply go/vtbench/vtbench.go]: Whoops, thanks. Removed.

**Comment by demmer** [comment]
@sougou I addressed Rafael's comments and added a usage example. Can you let me know if this is what you had in mind before I merge?

**Pull request merged**